include/tscore/ts_file.h

```diff
@@ -0,0 +1,245 @@
+     * @param that Filesystem path.
+     * @return @a this
+     */
+    self_type &operator/=(const self_type &that);
+    self_type &operator/=(std::string_view that);
+
+    /// Check if the path is empty.
+    bool empty() const;
+
+    /// Check if the path is absolute.
+    bool is_absolute() const;
+
+    /// Check if the path is not absolute.
+    bool is_relative() const;
+
+    /// Access the path explicitly.
+    char const *c_str() const;
+
+    /// Get a copy of the path.
+    std::string string() const;
+
+  protected:
+    std::string _path; ///< File path.
+  };
+
+  /// Information about a file.
+  class file_status
+  {
+    using self_type = file_status;
+
+  public:
+  protected:
+    struct ::stat _stat; ///< File information.
+
+    friend self_type status(const path &, std::error_code &) noexcept;
+
+    friend int file_type(const self_type &);
+    friend off_t file_size(const self_type &);
+    friend bool is_regular_file(const file_status &);
+    friend bool is_dir(const file_status &);
+    friend bool is_char_device(const file_status &);
+    friend bool is_block_device(const file_status &);
+  };
+
+  /** Get the status of the file at @a p.
+   *
+   * @param p Path to file.
+   * @param ec Error code return.
+   * @return Status of the file.
+   */
+  file_status status(const path &p, std::error_code &ec) noexcept;
+
+  // Related free functions.
+  // These are separate because they are not part of std::filesystem::path.
+
+  /// Return the file type value.
+  int file_type(const file_status &fs);
+
+  /// Check if the path is to a regular file.
+  bool is_regular_file(const file_status &fs);
+
+  /// Check if the path is to a directory.
+  bool is_dir(const file_status &p);
+
+  /// Check if the path is to a character device.
+  bool is_char_device(const file_status &fs);
+
+  /// Check if the path is to a block device.
+  bool is_block_device(const file_status &fs);
+
+  /// Size of the file or block device.
+  off_t file_size(const file_status &fs);
+
+  /// Check if file is readable.
+  bool is_readable(const path &s);
+
+  /** Load the file at @a p into a @c std::string.
+   *
+   * @param p Path to file
+   * @return The contents of the file.
+   */
+  std::string load(const path &p, std::error_code &ec);
+  /* ------------------------------------------------------------------- */
+
+  inline path::path(char const *src) : _path(src) {}
+
+  inline path::path(std::string_view base) : _path(base) {}
+
+  inline path::path(std::string &&that) : _path(std::move(that)) {}
+
+  inline path &
+  path::operator=(std::string_view p)
+  {
+    _path.assign(p);
+    return *this;
+  }
+
+  inline char const *
+  path::c_str() const
+  {
+    return _path.c_str();
+  }
+
+  inline std::string
+  path::string() const
+  {
+    return _path;
+  }
+
+  inline bool
+  path::empty() const
+  {
+    return _path.empty();
+  }
+
+  inline bool
+  path::is_absolute() const
+  {
+    return !_path.empty() && '/' == _path[0];
```

Comment:
Should use the preferred_separator or SEPERATOR or have a _is_seperator function/macro